### PR TITLE
Problem: not allowed specify nonce when estimate gas

### PR DESCRIPTION
**Pull request by mmsqe**
Closes: #XXX

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: x/evm/keeper/grpc_query.go [hunk 1]
```diff
@@ -333,9 +333,11 @@ func (k Keeper) EstimateGas(c context.Context, req *types.EthCallRequest) (*type
 		return nil, status.Error(codes.Internal, "failed to load evm config")
 	}
 
-	// ApplyMessageWithConfig expect correct nonce set in msg
-	nonce := k.GetNonce(ctx, args.GetFrom())
-	args.Nonce = (*hexutil.Uint64)(&nonce)
+	if args.Nonce == nil {
+		// ApplyMessageWithConfig expect correct nonce set in msg if not specify in request
+		nonce := k.GetNonce(ctx, args.GetFrom())
+		args.Nonce = (*hexutil.Uint64)(&nonce)
```
Comment (mmsqe) [inline x/evm/keeper/grpc_query.go:339]: [hermes](https://github.com/devashishdxt/hermes/pull/7/files) bump up nonce when call estimate

Comment (yihuang) [reply x/evm/keeper/grpc_query.go:339]: isn't the nonce returned by `k.GetNonce` is most correct?

Comment (mmsqe) [reply x/evm/keeper/grpc_query.go:339]: nonce could be the same from `k.GetNonce` when estimate batches, 2nd batch call before 1st batch finalize.

Comment (yihuang) [reply x/evm/keeper/grpc_query.go:339]: is it possible to describe the test case in integration test outside of the ibc context, still confused what are you trying to do.

Comment (mmsqe) [reply x/evm/keeper/grpc_query.go:339]: just make it easier for me to compare between client and server, otherwise I need compare with data field ;D

**Pull request closed without merging**